src/transformStateWithClones.js

```diff
@@ -6,8 +6,44 @@
  *
  * @return {Object[]}
  */
+function stateClear(state) {
+  for (const prop in state) {
+    delete state[prop];
+  }
+}
+
+function stateRemoveProperty(state, keysArr) {
+  for (const key of keysArr) {
+    delete state[key];
+  }
+}
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArray = [];
+  const currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(currentState, action.extraData);
```

Comment:
if you started creating separate functions for actions, why did you skip `addProperties`?
I know it is only one line but better to move it too

Reply:
This function would do the same as Object.assign. IMHO, you should create "one-line" functions when we have specific case of method (for example, sqrt() is a specific case of pow() ). But that function would do exactly same thing: assign properties from one object to another.